### DISCUSSION TITLE
How to use in combination with GraphQL Mesh

### DISCUSSION THREAD

**Discussion by paales** [dotansimha/graphql-code-generator#4015] [General] [answered]
Hi there! Great software, really appreciated! 

We're building a NextJS website where we have created a /api/graphql endpoint with GraphQL Mesh. We've build the application in typescript.

I can programmatically export the schema with the following typescript code:
```ts
const { schema } = (async () => getMesh(await findAndParseConfig()))()
```

Is there a way to hook this into Graphql Code Generator? Of course I'd like to keep the codegen.yml file instead of resorting to some programmatic method

**Comment by ardatan** [comment, marked as answer]
GraphQL Mesh gives you a regular `GraphQLSchema` which can be used in GraphQL Code Generator without any extra configuration.

You have two options;

First, you can just point to the URL
```
schema: http://localhost:3000/api/graphql
```

**### OR**

Second, you can export `schema` and point to that file in `codegen.yml`.

**schema.ts**
```
const { schema } = await getMesh(...);
export default schema;
```

and in **codegen.yml**
```
require: ts-node/register
schema: ./schema.ts
...
```

**Reply by ajibadeabd** [reply]
this was also helpful while integrating with nest.js

**Comment by paales** [comment]
I've got it working, had to do something like this, because my modules aren't in commonjs

```
TS_NODE_COMPILER_OPTIONS='{"module":"commonjs"}' yarn graphql-codegen
```

In my codegen.yml

```
require:
  - ts-node/register/transpile-only
  - tsconfig-paths/register
schema: ./node/meshSchema.ts
```

Thanks a lot for pushing me in the right direction!
